### PR TITLE
Move two tests from `tests/ui/std` to `library/std/tests`

### PR DESCRIPTION
**Pull request by 766974616c79**
Hi, there,
This pull request comes from this issue (#99417), sorry I made some mistakes creating the pull request, it's my first one.

### CODE REVIEW

File: library/std/tests/switch-stdout.rs [hunk 1]
```diff
@@ -35,16 +32,18 @@ fn switch_stdout_to(file: File) {
     }
 }
 
-fn main() {
-    let path = PathBuf::from(env::var_os("RUST_TEST_TMPDIR").unwrap());
-    let path = path.join("switch-stdout-output");
+#[test]
+fn switch_stdout() {
```
[the remark this answers is not in the document]
Comment (MoskalykA) [reply library/std/tests/switch-stdout.rs:38]: You're right, it was even written at the beginning of the file, thank you <3

File: library/std/tests/issue-15149.rs
Comment (workingjubilee) [inline library/std/tests/issue-15149.rs:1]: Can you give this a better name that describes what it is testing?

Comment (MoskalykA) [reply library/std/tests/issue-15149.rs:1]: Do you mean the file name or something else?

Comment (workingjubilee) [reply library/std/tests/issue-15149.rs:1]: Yes, the filename. `issue-{N}` is kind of meaningless if you aren't actively browsing through https://github.com/rust-lang/rust/issues at that precise moment, and while I personally do a lot of that, it's a bit of a context switch, so it's nice to see a semi-meaningful name at-a-glance.

Comment (MoskalykA) [reply library/std/tests/issue-15149.rs:1]: What do you think of a name like `process_spawning.rs`?

Comment (workingjubilee) [reply library/std/tests/issue-15149.rs:1]: Sure, sounds fine to me!

**Comment by workingjubilee** [comment]
Yes, "ready" specifically.

File: library/std/tests/switch-stdout.rs [hunk 2]
```diff
@@ -1,10 +1,9 @@
-// run-pass
-// ignore-wasm (needs file descriptors and env variables)
+#[cfg(any(target_family = "unix", target_family = "windows"))]
```
Comment (kadiwa4) [inline library/std/tests/switch-stdout.rs:1]: You meant to write `#!`

Comment (raoulstrackx) [reply library/std/tests/switch-stdout.rs:1]: Submitted a PR for this

Comment (MoskalykA) [reply library/std/tests/switch-stdout.rs:1]: Why is this? I want to ignore all the other systems

Comment (raoulstrackx) [reply library/std/tests/switch-stdout.rs:1]: A `#[cfg(...)]` is a configuration specification for the next line. So in this case it says something about the `use std::fs::File;` (Yes the new line is confusing here, which is why it slipped through review.).
A `#![cfg(...)]` says something about the entire module. That's what you want. It's pure coincidence that this wasn't picked up by CI either.

Comment (workingjubilee) [reply library/std/tests/switch-stdout.rs:1]: Oof, sorry for missing that.

Comment (MoskalykA) [reply library/std/tests/switch-stdout.rs:1]: I'm also sorry 😕

Comment (raoulstrackx) [reply library/std/tests/switch-stdout.rs:1]: Writing software is hard. That's why we run things in CI :)

**Pull request merged**